src/plugin/index-links/src/java/org/apache/nutch/indexer/links/LinksIndexingFilter.java

```diff
@@ -0,0 +1,168 @@
+import org.apache.nutch.crawl.Inlink;
+import org.apache.nutch.crawl.Inlinks;
+import org.apache.nutch.indexer.IndexingException;
+import org.apache.nutch.indexer.IndexingFilter;
+import org.apache.nutch.indexer.NutchDocument;
+import org.apache.nutch.parse.Outlink;
+import org.apache.nutch.parse.Parse;
+import org.slf4j.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+/**
+ * An {@link org.apache.nutch.indexer.IndexingFilter} that adds
+ * <code>outlinks</code> and <code>inlinks</code> field(s) to the document.
+ *
+ * In case that you want to ignore the outlinks that point to the same host
+ * as the URL being indexed use the following settings in your configuration
+ * file:
+ *
+ * <property>
+ *   <name>outlinks.host.ignore</name>
+ *   <value>true</value>
+ * </property>
+ *
+ * The same configuration is available for inlinks:
+ *
+ * <property>
+ *   <name>inlinks.host.ignore</name>
+ *   <value>true</value>
+ * </property>
+ *
+ * To store only the host portion of each inlink URL or outlink URL add the
+ * following to your configuration file.
+ *
+ * <property>
+ *   <name>links.hosts.only</name>
+ *   <value>false</value>
+ * </property>
+ *
+ */
+public class LinksIndexingFilter implements IndexingFilter {
+
+  public final static String LINKS_OUTLINKS_HOST = "outlinks.host.ignore";
+  public final static String LINKS_INLINKS_HOST = "inlinks.host.ignore";
+  public final static String LINKS_ONLY_HOSTS = "links.hosts.only";
+
+  public final static org.slf4j.Logger LOG = LoggerFactory
+      .getLogger(LinksIndexingFilter.class);
+
+  private Configuration conf;
+  private boolean filterOutlinks;
+  private boolean filterInlinks;
+  private boolean indexHost;
+
+  @Override
+  public NutchDocument filter(NutchDocument doc, Parse parse, Text url,
+      CrawlDatum datum, Inlinks inlinks) throws IndexingException {
+
+    // Add the outlinks
+    Outlink[] outlinks = parse.getData().getOutlinks();
+
+    try {
+      if (outlinks != null) {
+        Set<String> hosts = new HashSet<String>();
+
+        for (Outlink outlink : outlinks) {
+          String linkUrl = outlink.getToUrl();
+          String outHost = new URL(linkUrl).getHost();
+
+          if (indexHost) {
+            linkUrl = new URL(outlink.getToUrl()).getHost();
+
+            if (hosts.contains(linkUrl))
+              continue;
+
+            hosts.add(linkUrl);
+          }
+
+          addFilteredLink("outlinks", url.toString(), linkUrl, outHost,
+              filterOutlinks, doc);
+        }
+      }
+    } catch (MalformedURLException e) {
+      LOG.error("Malformed URL in {}: {}", url, e.getMessage());
+    }
+
+    // Add the inlinks
+    if (null != inlinks) {
+      Iterator<Inlink> iterator = inlinks.iterator();
+      Set<String> inlinkHosts = new HashSet<String>();
+
+      try {
+        while (iterator.hasNext()) {
```

Comment:
+1 great!

On 10/15/2015 09:20 PM, Jorge Luis Betancourt wrote:
> In src/plugin/index-links/src/java/org/apache/nutch/indexer/links/LinksIndexingFilter.java
> <https://github.com/apache/nutch/pull/78#discussion_r42168120>:
> 
>> +
>> +          addFilteredLink("outlinks", url.toString(), linkUrl, outHost,
>> +              filterOutlinks, doc);
>> +        }
>> +      }
>> +    } catch (MalformedURLException e) {
>> +      LOG.error("Malformed URL in {}: {}", url, e.getMessage());
>> +    }
>> +
>> +    // Add the inlinks
>> +    if (null != inlinks) {
>> +      Iterator<Inlink> iterator = inlinks.iterator();
>> +      Set<String> inlinkHosts = new HashSet<String>();
>> +
>> +      try {
>> +        while (iterator.hasNext()) {
> 
> I though the same, since the URL is already is fetched shouldn't be any trouble, but its an easy fix
> so I can put it inside the while loop.
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/nutch/pull/78/files#r42168120>.
>